File: core/src/main/java/com/codeabovelab/tpc/core/text/TextCoordinates.java
package com.codeabovelab.tpc.core.text;

/**
 * Text coordinates in {@link Textual } implementation.
 * Usually it offset and length, but for images or multi page document it may be more complex.
 */
public interface TextCoordinates {
    /**
     * If of textual object which contains this coordinates.
     * @return non null string
     */
    String getTextualId();

    /**
     * Offset from document begin.
     * @return offset
     */
    int getOffset();

    /**
     * Length, can be 0.
     * @return length
     */
    int getLength();
}

File: common/src/main/java/com.codeabovelab.tpc/text/TextCoordinates.kt
package com.codeabovelab.tpc.text

/**
 * Text coordinates in [Textual] implementation.
 * Usually it offset and length, but for images or multi page document it may be more complex.
 */
interface TextCoordinates {
    /**
     * Offset from document begin.
     * @return offset
     */
    val offset: Int

    /**
     * Length, can be 0.
     * @return length
     */
    val length: Int
}
